### PR TITLE
Filter empty group items

### LINKED ISSUE #1753: WS Preprocessor - Filter out empty sections (groups)
To avoid any issues with CPS feeds, to disable a section on the page, we have decided the journalist can just remove all stories from its section.

**Describe the solution you'd like**
Taking in CPS homepage index data, create a preprocessor rule that deletes groups from the data if they contain 0 items

An example of a preprocessor rule can be found [here](https://github.com/bbc/simorgh/blob/9fa9eb596a3eae1d6d6d1e87eab344e9ee9db472/src/app/lib/utilities/preprocessor/rules/topstories.js). TL;DR, take in json, split out json 

**Testing notes**
[Tester to complete]

Dev insight: Unit tests

### PR DESCRIPTION
**Pull request by ghost**
Resolves #1753 

**Overall change:** Added preprocessor rule that deletes groups from the data if they contain 0 items

**Code changes:**

- Added a function to filter out `groups` with `0` items.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/groups.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const data = {
+      content: {
+        groups: [
+          {
+            type: 'top-story',
+            items: [{ name: 'item1' }],
+          },
+          {
+            type: 'must-see',
+            items: [],
```
Comment (dr3) [inline src/app/lib/utilities/preprocessor/rules/groups.test.js]: Can you please add support for items being undefined, null or a string please. Unlikely to ever happen but good to be safe :)

File: src/app/lib/utilities/preprocessor/rules/groups.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import deepGet from '../../../../helpers/json/deepGet';
+
+/* eslint-disable import/prefer-default-export */
+export const filterEmptyGroupItems = jsonRaw => {
```
Comment (dr3) [inline src/app/lib/utilities/preprocessor/rules/groups.js]: This can be a single rule, named with it being a default export :)

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/lib/utilities/preprocessor/rules/groups.js (hunk 2 above)
Comment (ghost) [reply src/app/lib/utilities/preprocessor/rules/groups.js]: I was thinking more like. We might have other `rules` for groups

Comment (dr3) [reply src/app/lib/utilities/preprocessor/rules/groups.js]: Yeah, I think each one in its own file is fine, nice and easy to see each rule in isolation IMO

**Review by dr3 — DISMISSED** [review]
LGTM, nice work

**Review by DenisHdz — DISMISSED** [review]
LGTM  :)

**Comment by david-boydell** [comment]
Unit tests cover this for the moment, it's RFM.

**Comment by dr3** [comment]
@sadickisaac Can you check if theres any way we can increase the test coverage on your change? It says it only has 88% coverage so we cant merge

**Review by dr3 — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]

**Comment by david-boydell** [comment]
No manual testing here, this is RFM with verification later when everything's hooked up.

**Pull request merged**